src/main/java/de/rwth/idsg/steve/repository/impl/WebUserRepositoryImpl.java

```diff
@@ -0,0 +1,226 @@
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.provisioning.JdbcUserDetailsManager;
+import org.springframework.stereotype.Repository;
+import org.springframework.util.Assert;
+
+import java.util.Collection;
+import java.util.LinkedHashSet;
+import java.util.stream.Collectors;
+
+import static jooq.steve.db.Tables.WEB_USER;
+import static org.jooq.impl.DSL.condition;
+import static org.springframework.security.authentication.UsernamePasswordAuthenticationToken.authenticated;
+import static org.springframework.security.core.context.SecurityContextHolder.getContextHolderStrategy;
+
+/**
+ * Inspired by {@link org.springframework.security.provisioning.JdbcUserDetailsManager}
+ *
+ * @author Sevket Goekay <sevketgokay@gmail.com>
+ * @since 10.08.2024
+ */
+@Slf4j
+@Repository
+@RequiredArgsConstructor
+public class WebUserRepositoryImpl implements WebUserRepository {
+
+    private final DSLContext ctx;
+    private final ObjectMapper jacksonObjectMapper;
+    private final SecurityContextHolderStrategy securityContextHolderStrategy = getContextHolderStrategy();
+
+    @Override
+    public void createUser(UserDetails user) {
+        validateUserDetails(user);
+
+        ctx.insertInto(WEB_USER)
+            .set(WEB_USER.USERNAME, user.getUsername())
+            .set(WEB_USER.PASSWORD, user.getPassword())
+            .set(WEB_USER.ENABLED, user.isEnabled())
+            .set(WEB_USER.AUTHORITIES, toJson(user.getAuthorities()))
+            .execute();
+    }
+
+    @Override
+    public void updateUser(UserDetails user) {
+        validateUserDetails(user);
+
+        ctx.update(WEB_USER)
+            .set(WEB_USER.PASSWORD, user.getPassword())
+            .set(WEB_USER.ENABLED, user.isEnabled())
+            .set(WEB_USER.AUTHORITIES, toJson(user.getAuthorities()))
+            .where(WEB_USER.USERNAME.eq(user.getUsername()))
+            .execute();
+    }
+
+    @Override
+    public void deleteUser(String username) {
+        ctx.delete(WEB_USER)
+            .where(WEB_USER.USERNAME.eq(username))
+            .execute();
+    }
+
+    @Override
+    public void deleteUser(int webUserPk) {
+        ctx.delete(WEB_USER)
+            .where(WEB_USER.WEB_USER_PK.eq(webUserPk))
+            .execute();
+    }
+
+    @Override
+    public void changeStatusOfUser(String username, boolean enabled) {
+        ctx.update(WEB_USER)
+            .set(WEB_USER.ENABLED, enabled)
+            .where(WEB_USER.USERNAME.eq(username))
+            .execute();
+    }
+
+    @Override
+    public boolean hasUserWithAuthority(String authority) {
+        JSON authValue = JSON.json("\"" + authority + "\"");
+        return ctx.selectOne()
+            .from(WEB_USER)
+            .where(condition("json_contains({0}, {1})", WEB_USER.AUTHORITIES, authValue))
+            .fetchOptional()
```

Comment:
fetchOptional(): use this only if the query returns at most one record. 
So if there a 2 or more admins it crashes. 
--> .fetch().isNotEmpty()

Reply:
good catch, thanks.

Reply:
fixed in [0992d5c](https://github.com/steve-community/steve/pull/1539/commits/0992d5cdc3f13ead84e1d09157b8d1bfdd80bfde)

i opted for count, since we dont need all these rows with 1. count is a better fitting usage.